utils/generate_structures.py

```diff
@@ -1,74 +1,58 @@
 from typing import List
 from lsp_schema import Structure
-from utils.helpers import FormattedProperty, format_comment, indentation, format_class_properties, format_dict_properties, get_formatted_properties, has_invalid_property_name
+from utils.helpers import FormattedProperty, format_comment, indentation, format_class_properties, format_dict_properties, get_formatted_properties, has_invalid_property_name, StructureKind
 
 
-def generate_structures(structures: List[Structure]) -> List[str]:
+def generate_structures(structures: List[Structure], preferred_structure_kind: StructureKind) -> List[str]:
 
     def toString(structure: Structure) -> str:
-        if has_invalid_property_name(structure['properties']):
-            return generate_function_type(structure, structures)
-        return generate_class_type(structure, structures)
+        kind = preferred_structure_kind
+        if kind == StructureKind.Class and has_invalid_property_name(structure['properties']):
+            kind = StructureKind.Function
+        return generate_structure(structure, structures, kind)
 
     return [toString(structure) for structure in structures]
 
 
-def get_additional_properties(for_structure: Structure, structures: List[Structure]) -> List[FormattedProperty]:
-    """Returns properties from extended and mixin types. """
-    result: List[FormattedProperty] = []
-    additional_structures = for_structure.get('extends') or []
-    additional_structures.extend(for_structure.get('mixins') or [])
-    for additional_structure in additional_structures:
-        if additional_structure['kind'] != 'reference':
-            raise Exception("Cannot generate extends. Currently only supports kind: 'reference', but received:", additional_structure['kind'])
-        structure = next(structure for structure in structures if structure["name"] == additional_structure['name'])
-        if structure:
-            properties = get_formatted_properties(structure['properties'])
-            result.extend(properties)
-    return result
-
-
-def generate_class_type(structure: Structure, structures: List[Structure]) -> str:
+def generate_structure(structure: Structure, structures: List[Structure], structure_kind: StructureKind) -> str:
```

Comment:
`generate_class_type` and `generate_function_type` were almost identical so combined and added a `structure_kind` argument.